multinet/db.py

```diff
@@ -202,11 +202,17 @@ def workspace_table_keys(
         LIMIT 0, 1
         RETURN ATTRIBUTES(d)
     """
-    cursor = next(aql_query(workspace, query))
+    cur = aql_query(workspace, query)
+
+    try:
+        keys = next(cur)
+    except StopIteration:
+        return []
+
     if filter_keys:
-        return [k for k in cursor if k not in restricted_keys]
+        return [k for k in keys if k not in restricted_keys]
 
-    return list(aql_query(workspace, query))
+    return list(cur)
```

Comment:
Shouldn't this be 

```suggestion
    return keys
```

Since you ran the cursor already in like 208, if `filter_keys` isn't `True`, won't this just return an empty list?